tests/unit/symbolic_regression/test_symbolic_regressor.py

```diff
@@ -1,216 +1,565 @@
-    assert regr.evolutionary_algorithm == mocked_ea
-    assert regr.get_params()["evolutionary_algorithm"] == mocked_ea
+    island = regr._make_island(dataset_size, sample_ea, hof)
 
-    assert regr.population_size == 1
-    assert regr.get_params()["population_size"] == 1
+    assert isinstance(island, expected_island)
+    assert island._ea == sample_ea
+    assert island._generator == sample_agraph_gen
+    assert island._population_size == pop_size
+    assert island.hall_of_fame == hof
 
 
-def test_setting_invalid_param_raises_value_error():
+def test_force_diversity_in_island(sample_ea, sample_agraph_gen):
+    np.random.seed(0)
+    pop_size = 100
+    island = Island(sample_ea, sample_agraph_gen, pop_size)
     regr = SymbolicRegressor()
-    with pytest.raises(ValueError):
-        regr.set_params(**{"not_a_valid_param": False})
+    regr.population_size = pop_size
+    regr.generator = sample_agraph_gen
 
+    regr._force_diversity_in_island(island)
 
-def test_get_best_individual(mocker):
+    unique_indvs = set(str(indv) for indv in island.population)
+    assert len(unique_indvs) == pop_size
+    assert len(island.population) == pop_size
+
+
+@pytest.fixture
+def sample_agraph():
+    equation = AGraph()
+    equation.command_array = np.array([[0, 0, 0]], dtype=int)
+    return equation
+
+
+@pytest.mark.timeout(3)
+def test_cant_force_diversity_in_island(mocker, sample_ea, sample_agraph):
+    np.random.seed(0)
+    pop_size = 100
+    agraph_gen = mocker.Mock(return_value=sample_agraph)
+    island = Island(sample_ea, agraph_gen, pop_size)
     regr = SymbolicRegressor()
-    best_ind = mocker.Mock()
-    regr.best_ind = best_ind
+    regr.population_size = pop_size
+    regr.generator = agraph_gen
 
-    assert regr.get_best_individual() == best_ind
+    regr._force_diversity_in_island(island)
+
+    unique_indvs = set(str(indv) for indv in island.population)
+    assert len(unique_indvs) == 1
+    assert len(island.population) == pop_size
+
+
+def patch_all_get_archipelago(mocker):
+    imports = ["ComponentGenerator", "AGraphCrossover", "AGraphMutation",
+               "AGraphGenerator", "Evaluation", "AgeFitnessEA",
+               "DeterministicCrowdingEA", "HallOfFame"]
+    fns = ["_get_local_opt", "_make_island", "_force_diversity_in_island"]
+
+    patched_imports = {name: patch_import(mocker, name) for name in imports}
+    patched_fns = {name: patch_regr_method(mocker, name) for name in fns}
+
+    patched_imports.update(patched_fns)
+
+    return patched_imports
+
+
+def get_mocked_data(mocker):
+    len = 250
+    X = mocker.MagicMock()
+    X.__len__.return_value = len
+    X.shape = (len, mocker.Mock())
+
+    y = mocker.MagicMock()
+    y.__len__.return_value = len
+    y.shape = (len, mocker.Mock())
+
+    return X, y
+
+
+@pytest.mark.parametrize("operators", [None, "mock"])
+def test_get_archipelago_component_gen(mocker, operators):
+    patched_objs = patch_all_get_archipelago(mocker)
```

Comment:
This is probably not the best way to be isolating components for testing, but it is very repetitive to mock only the objects necessary for each test case. If you think there's a better way, let me know.